docs/consume-packages/managing-the-global-packages-and-cache-folders.md

```diff
@@ -0,0 +1,116 @@
+---
+title: How to manage the global packages, cache, temp folders in NuGet | Microsoft Docs
+author: kraigb
+ms.author: kraigb
+manager: ghogen
+ms.date: 03/19/2018
+ms.topic: article
+ms.prod: nuget
+ms.technology: null
+description: How to manage the global package installation folder, the package cache, and the temp folders that exist on a computer, which are used when installing, restoring, and updating packages.
+keywords: NuGet global packages folder, NuGet package cache, package caching, package installation folder, NuGet caches, managing caches, local NuGet cache, global NuGet cache, NuGet locals command, clearing a cache
+ms.reviewer:
+- karann-msft
+- unniravindranathan
+---
+
+# Managing the global packages, cache, and temp folders
+
+Whenever you install, update, or restore a package, NuGet manages packages and package information in several folders outside of your project structure:
+
+| Name | Description and Location (per user)|
+| --- | --- |
+| global&#8209;packages | The *global-packages* folder is where NuGet installs any downloaded package. Each package is fully expanded into a subfolder that matches the package identifier and version number. Projects using the PackageReference format always use packages directly from this folder. When using the `packages.config`, packages are installed to the *global-packages* folder, then copied into the project's `packages` folder.<br/><ul><li>Windows: `%userprofile%\.nuget\packages`</li><li>Mac/Linux: `~/.nuget/packages`</li><li>Override using the NUGET_PACKAGES environment variable, or the `globalPackagesFolder` [configuration setting](../reference/nuget-config-file.md#config-section) (PackageReference only). The environment variable takes precedence over the configuration setting.</li></ul> |
+| http&#8209;cache | The Visual Studio Package Manager (NuGet 3.x+) and the `dotnet` tool store copies of downloaded packages in this cache (saved as `.dat` files), organized into subfolders for each package source. Packages are not expanded, and the cache is not used by the `nuget.exe` tool. The cache has an expiration time of 30 minutes.<br/><ul><li>Windows: `%localappdata%\NuGet\v3-cache`</li><li>Mac/Linux: `~/.local/share/NuGet/v3-cache`</li><li>Override using the NUGET_HTTP_CACHE_PATH environment variable.</li></ul> |
```

Comment:
nuget.exe also uses the same http cache which is why we support -NoCache and -DirectDownload flags with nuget.exe.

Reply:
So does it draw from the cache when installing for packages, but not save downloaded packages in the cache? I'm not observing that. For example, if I go to some folder and nuget install a package, then delete the folder where the package was just installed, and also delete the package from global-packages, then nuget install the same package again, I see the HTTP GET and OK in the output; I do not see CACHE, and indeed, the package isn't even in the cache.

Even if I dotnet add package, which does write it to the cache, then still, nuget install shows HTTP GET and OK in the output.

What's happening here that I don't understand properly?

Reply:
PS, I'll remove that line for now while we resolve the matter.